### ISSUE TITLE
[ Repo auth ] Fails to build a byte code repository on hhvm 4.168+

### ISSUE THREAD

**Issue by lexidor** [facebook/hhvm#9236]
**Describe the bug**
When trying to build a byte code repository in docker, hhvm 4.168 fails with a `set_mempolicy: Operation not permitted`

**Standalone code, or other way to reproduce the problem**
```HACK
<<__EntryPoint>>
function main(): void {}
```

```INI
hhvm.repo.authoritative=true
hhvm.repo.path=./hhvm.hhbc
hhvm.server.allow_run_as_root=true
```

1. Create an empty Hack project in an hhvm/hhvm docker container
2. Add an empty hhconfig file.
3. Put an empty entry point, as shown above in src/main.hack
4. Put the INI file shown above in repo-auth.ini
5. Create a repo using `hhvm --hphp --input-dir $(pwd) --output-dir .`
6. When running under hhvm/hhvm@4.167-latest, observe a successful creation of hhvm.hhbc in the project directory.
7. Run the server using `hhvm -m server --no-config -c repo-auth.ini`
8. Curl (with option verbose flag) localhost from within the container. Observe an HTTP 200 without content.

**Expected behavior**

As described in steps to reproduce

**Actual behavior**

Step 5. fails with the following error:
```
...
Extern worker error while indexing: Execution of `hphpc-index` failed: exited with status 1
stderr:
set_mempolicy: Operation not permitted
...
```

**Environment**
 - Operating system
> Host system Ubuntu 22.04.
 - Installation method
> hhvm/hhvm on dockerhub
 - HHVM Version
> 4.168.0, as provided in the docker container

**Additional context**
I did already change the commands to be compatible with the removed cli flags from hhvm 4.167.0.

**Comment by Atry** [comment]
Is it a docker specific problem? Did you test repo mode without a docker container?

**Comment by Atry** [comment]
Looks similar to the CI failures in https://github.com/hhvm/hacktest/pull/115/files

**Comment by Atry** [comment]
This is not a docker specific problem. I can reproduce it without a docker container.

**Comment by Atry** [comment]
I can also reproduce it with the nix based universal HHVM deb package

**Comment by Atry** [comment]
I cannot reproduce this issue with a HHVM built from clang.

**Comment by Atry** [comment]
As a workaround, you can use the HHVM package built from clang to use repo auth. 

You can find the instruction to install `hhvm_clang` packages at https://github.com/hhvm/user-documentation/pull/1295

**Comment by Atry** [comment]
The nightly version 2022.09.01 is good to use repo mode, while the nightly version 2022.09.02 is not.

The failure seems due to 280e6899b442796a197b5b236d3ddd50d0b62b5f.

**Comment by Atry** [comment]
Fixed in 1174b9f2e5935ff468b236aba9df0f3d3ff12d5a
